### PR TITLE
[MarketplaceOrdering] New command group `az term` to accept/show terms

### PR DESCRIPTION
**Pull request by yungezz**
**Description**

MarketplaceOrdering RP  https://docs.microsoft.com/en-us/rest/api/marketplaceordering/marketplaceagreements, API version 2021-01-01, on Track2 SDK. CodeGen readme config file: https://github.com/Azure/azure-rest-api-specs/pull/12484/files

- `az term accept` covers term accpeting operation, calling into API PUT https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.MarketplaceOrdering/offerTypes/virtualmachine/publishers/{publisherId}/offers/{offerId}/plans/{planId}/agreements/current?api-version=2021-01-01
- `az term show` covers term show operation, calling into API GET https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.MarketplaceOrdering/offerTypes/virtualmachine/publishers/{publisherId}/offers/{offerId}/plans/{planId}/agreements/current?api-version=2021-01-01
- `offerType` is default `virtualmachine`, this is behavior confirmed with Marketplace team.

**Why this is a core module instead of extension?**
Terms accepting is a general flow in mgmt plane e2e flow. It's a cross RP operation. Accepting term is a key step during RP registration, user will not able to use RP like LiftR specific commands without terms. Put it in core improving discoverability and availability to users. Secondly,  marketplaceordering SDK is already a dependency in core repo from long time ago, this will not impact core package heavily.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by fengzhou-msft** [comment]
Should we use `az term` or `az terms`? It looks like `terms` is more common as a command name or option name.

**Comment by yungezz** [comment]
> Should we use `az term` or `az terms`? It looks like `terms` is more common as a command name or option name.
use `term` to follow existing command group naming convention: single noun like `vm`, `acr`, `keyvault` etc. what's your thinking?

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+
+from knack.help_files import helps
+
+
+helps['marketplaceordering'] = '''
+    type: group
+    short-summary: Manage Marketplace Ordering Agreements
+'''
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py:18]: This should be gone.

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_params.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# pylint: disable=too-many-statements
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('term show') as c:
+        c.argument('publisher', type=str, help='Publisher identifier string of image being deployed.',
+                   id_part='child_name_1')
+        c.argument('offer', type=str, help='Offer identifier string of image being deployed.', id_part='child_name_2')
+        c.argument('plan', type=str, help='Plan identifier string of image being deployed.', id_part='child_name_3')
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_params.py:20]: I think `--ids` will fail without the `name` id_part.

**Comment by fengzhou-msft** [comment]
> > Should we use `az term` or `az terms`? It looks like `terms` is more common as a command name or option name.
> 
> use `term` to follow existing command group naming convention: single noun like `vm`, `acr`, `keyvault` etc. what's your thought?

Yes, it's more common to have single form noun as command group. I was think that it is more common to say `accept terms` than `accept a term`. For command group name, I think both forms are acceptable.

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py (hunk 1 above)
Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py:18]: Will file an issue to codegen

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_params.py (hunk 2 above)
Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_params.py:20]: feedback to codegen also, the team is working on fixing.

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py (hunk 1 above)
Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py:18]: https://github.com/Azure/autorest.az/issues/825

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+helps['term'] = """
+    type: group
+    short-summary: Manage marketplace agreement with marketplaceordering
+"""
+
+helps['term show'] = """
+    type: command
+    short-summary: "Get marketplace terms."
+    examples:
+      - name: GetMarketplaceTerms
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/_help.py:29]: Example name needs to be updated.

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/commands.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+
+def load_command_table(self, _):
+
+    with self.command_group(
+        'term', marketplaceordering_marketplace_agreement, client_factory=cf_marketplace_agreement, is_experimental=True
+    ) as g:
+        g.custom_show_command('show', 'term_show')
+        g.custom_command('accept', 'term_accept')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/commands.py:31]: How to cancel a marketplace term?

Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/marketplaceordering/generated/commands.py:31]: cancel via portal, so far service doesn't expose api to client to cancel except portal

**Review by fengzhou-msft — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/_help.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=wildcard-import
+# pylint: disable=unused-wildcard-import
```
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/marketplaceordering/_help.py:11]: You may need to import help info 
`from .generated._help import helps`

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/__init__.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+
+from azure.cli.core import AzCommandsLoader
+
```
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/marketplaceordering/__init__.py:12]: `import azure.cli.command_modules.marketplaceordering._help`

File: src/azure-cli/azure/cli/command_modules/marketplaceordering/_help.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+from knack.help_files import helps  # pylint: disable=unused-import
+
+from .generated._help import helps
+try:
+    from .manual._help import helps  # pylint: disable=reimported
+except ImportError:
+    pass
```
Comment (xfz11) [inline src/azure-cli/azure/cli/command_modules/marketplaceordering/_help.py:18]: This file should be generated by codegen. It seems that your generated file is different than mine.
```
from .generated._help import helps  # pylint: disable=reimported
try:
    from .manual._help import helps  # pylint: disable=reimported
except ImportError as e:
    if e.name.endswith('manual._help'):
        pass
    else:
        raise e
```

**Comment by qwordy** [comment]
@yungezz Example name needs to be updated. You might need to add a manual help file.

**Review by qwordy — APPROVED** [review]

**Pull request merged**